### ISSUE TITLE
[Feature]: Support datetime type 

### ISSUE THREAD

**Issue by xiaofan-luan** [milvus-io/milvus#27577]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem? Please describe.

Timestamp is a very common datatype in all databases.
We can convert timestamp into int64, but this will be a overhead for the user.

We can follow the definition of PG https://www.postgresql.org/docs/current/datatype-datetime.html about what datetime data type should behave

**Comment by fool1280** [comment]
Hi, I'm interested in tackling this issue. May I ask where should I start?

**Comment by xiaofan-luan** [comment]
Hi @fool1280 
1. start from adding a type from milvus.proto https://github.com/milvus-io/milvus-proto, there is a FieldSchema fields which you can 
2. implement create collection path, make sure we can create a collection with timestamp field
3. implement the insert path, this usually starts from proxy function  	Insert(ctx context.Context, in *InsertRequest, opts ...grpc.CallOption) (*MutationResult, error)
5. Implement the query/search path. make sure we can run where condition on datetime field

**Comment by xiaofan-luan** [comment]
> Thank you. I want to ask for the DateTime, would it be possible for me to use the `Timestamp` type in https://protobuf.dev/reference/protobuf/google.protobuf/#timestamp, as converting into int64 would introduce overhead for the user?

This is not too hard. you just need to convert date time to timestamp and there is no blocker

**Comment by dugar-tarun** [comment]
Hello @xiaofan-luan, I'm interested to take this up. Can you please assign this to me?

**Comment by xiaofan-luan** [comment]
/assign @dugar-tarun 
please let me know if you need any help

**Comment by yiwen92** [comment]
@dugar-tarun Hey, just check, how is this going?

**Comment by maheshpaliwal** [comment]
Looks like there is no activity on this issue, can you assign me ?

> This is not too hard. you just need to convert date time to timestamp and there is no blocker

did you mean create a separate data type for datetime instead of using timestamp ?

**Comment by xiaofan-luan** [comment]
sure, /assign @maheshpaliwal 

we do have some progress on timestamptz type，date time might be very similar to timestamptz

**Comment by Eelyousha** [comment]
Hello @xiaofan-luan I'd like to contribute on this one, can you please assign this one to me? Also what's current state of the feature?

**Comment by xiaofan-luan** [comment]
/assign @Eelyousha 
please take a look on it

**Comment by AyushKashyapII** [comment]
Hi @xiaofan-luan , just wanted to share a thought while reading through this :

time with time zone might not be worth implementing given PostgreSQL itself warns against it (DST can't be resolved without a date). Would it make sense to scope this down to just date, time without time zone, timestamp without time zone, and keep timestamptz since it's already in the enum?

Happy to pick up any part of this if the current assignees need help or the scope gets clarified.

**Comment by Eelyousha** [comment]
@AyushKashyapII gonna take this one 23.06 after about 2 PM GMT (now prep to some tech interview), so may connect to discuss after that time or so

**Comment by xiaofan-luan** [comment]
we'd happy to discuss all the details. 
I think aligned with pg is a smart approach.
any design docuemnts will be welcomed

**Comment by AyushKashyapII** [comment]
Hi @xiaofan-luan, here's a quick design note based on my reading of the pg spec:

Proposed types for this milestone:
Date → Date32 (days since epoch)
Time → Time64(us), no timezone
Timestamp → Timestamp(us), no timezone
Timestamptz → already in enum (26) 
Skip Time with timezone — pg itself warns against it, DST can't be resolved without a date
Filter literals: prefer ISO 8601 style (created_at > '2024-06-22') over SQL-style (DATE '2024-06-22') — shorter and consistent with what Milvus users already expect.

Also is Interval will be in scope so users could possibly query 
"give me all records created in the last 7 days"
created_at > now() - Interval("7 days")
On Interval: makes sense to me to defer it ,.it's more of a query arithmetic helper than a storage type, and the 4 types above cover the real storage use cases. Could be a natural follow-up PR once this lands.
cc: @Eelyousha 
source: https://www.postgresql.org/docs/current/datatype-datetime.html

**Comment by xiaofan-luan** [comment]
That sound to be very clear to be. 
start from date and time would be a good start

**Comment by Eelyousha** [comment]
Got it, I'm currently inspecting the project's code and how timestampz is implemented, next gonna make date and time then

**Comment by daviddallakyan2005** [comment]
@xiaofan-luan @Eelyousha @dugar-tarun

I am starting work on native DATE and TIME field types for this issue, scoped to the 2026-06-24 agreement. DATE as days since epoch (Date32, int32 storage). TIME without time zone as microseconds since midnight (Time64, int64 storage). ISO-8601 literals. Time with time zone is out of scope.

milvus-proto#621 already added `DataType.Date = 28` and `DataType.Time = 29`, plus DateArray and TimeArray. That pin is already in milvus go.mod, so I am not opening another proto PR. Next is a design doc under `docs/design-docs/design_docs/` and then the Go schema, typeutil, and proxy insert path, followed by C++ segcore and the in-tree Go client. pymilvus will be a separate repo.

Are the current assignees still active on the milvus implementation? I do not want to duplicate work if a PR is already in progress.

/cc @xiaofan-luan

**Comment by xiaofan-luan** [comment]
I don't think there is someone actively on it.

@tedxu can you please on coordinate on this issue as well?

**Comment by Eelyousha** [comment]
Hey, yeah, I'm working on it, will make PR at about 20.08 5 p.m. GMT, sorry for being too long, some life issues @daviddallakyan2005 @xiaofan-luan
